docs/adding_packages/dependencies.md

```diff
@@ -1,20 +1,181 @@
+However the [`self.dependencies`](https://docs.conan.io/en/latest/reference/conanfile/dependencies.html?highlight=generate) are available.
+
+Alternatively, a project may depend on a specific versions or configuration of a dependency. This use case is again covered by the
+[`self.dependencies`](https://docs.conan.io/en/latest/reference/conanfile/dependencies.html?highlight=validate) within the
+`validate()` method. Additionally it's possible to suggest the option's values while the graph is built through `configure()`
+this is not guaranteed and not a common practice.
+
+### Handling Requirement's Options
+
+Forcing options of dependencies inside a ConanCenter should be avoided, except if it is mandatory for the library to build.
+Our general belief is the users input should be the most important; it's unexpected for command line arguements to be over ruled
+by specifc recipes.
+
+You need to use the `validate()` method in order to ensure they check after the Conan graph is completely built.
+
+Certain projects are dependant on the configuration (a.k.a options) of a dependency. This can be enforced in a recipe by
+accessing the [`options`](https://docs.conan.io/en/latest/reference/conanfile/dependencies.html?highlight=options) feild of
+the dependency.
+
+```py
+  def configure(self):
+      self.options["foobar"].enable_feature = True # This will still allow users to override this option
+
+  def validate(self):
+      if not self.dependencies["foobar"].options.enable_feature:
+          raise ConanInvalidConfiguration(f"{self.ref} requires foobar/*:enable_feature=True.")
+```
+
+### Verifying Dependency's Version
+
+Some project requirements need to respect a version constraint, this can be done as follows:
+
+```py
+def validate(self):
+    if Version(self.dependencies["foobar"].ref.version) < "1.2":
+        raise ConanInvalidConfiguration(f"{self.ref} requires [foobar>=1.2] to build and work.")
+```
+
+### Passing Requirement's info to `build()`
+
+The `self.dependencies` are limtied to `generate()` and `validate()`. This means configuring a projects build scripts
+is a touch more complicated when working with unsupported build scripts.
+
+In general, with CMake project, this can be very simple with the `CMakeToolchain`, such as:
+
+```py
+    def generate(self):
+        tc = CMakeToolchain(self)
+        # deps_cpp_info, deps_env_info and deps_user_info are no longer used
+        if self.dependencies["dependency"].options.foobar:
+            tc.variables["DEPENDENCY_LIBPATH"] = self.dependencies["dependency"].cpp_info.libdirs
+```
+
+This pattern can be recreated for less common build system by, generating a script to call configure or  capture the
+required values in a YAML files for example.
+
+> **Note**: This needs to be saved to disk because the `conan install` and `conan build` commands can be seperated when
```

Comment:
```suggestion
> **Note**: This needs to be saved to disk because the [`conan install`](https://docs.conan.io/en/latest/reference/commands/consumer/install.html) and [`conan build`](https://docs.conan.io/en/latest/reference/commands/development/build.html) commands can be separated when
```